recipes/newrelic/infrastructure/cloud/aws-linux.yml

```diff
@@ -20,10 +20,103 @@ processMatch: []
 preInstall:
   requireAtDiscovery: |
       code=$(curl -I http://169.254.169.254/latest/meta-data/instance-id -w %{response_code} -so '/dev/null')
-      if [ $code == "200" ]; then
-        exit 132
+
+      if [ $code != "200" ]; then
+        exit 1
+      fi
+
+      if [ ! -f ~/.aws/credentials ]; then
+        exit 1
+      fi
+
+      if [ ! -f ~/.aws/config ]; then
+        exit 1
+      fi
+
+      if ! command -v aws &> /dev/null; then
+        exit 1
+      fi
+
+      callerIdentity=$(aws sts get-caller-identity 2>&1)
+
+      if [ "$(echo "$callerIdentity" | grep AccessDenied | wc -l)" -eq 1 ] ; then
+        exit 1
+      fi
+
+      function isUserAdmin {
+        username=$(echo "$1" | sed -ne 's/.*"Arn": ".*\/\(.*\)".*/\1/p')
+
+        userPolicies=$(aws iam list-user-policies --user-name $username 2>&1)
+
+        if [ "$(echo "$userPolicies" | grep AccessDenied | wc -l)" -eq 1 ] ; then
+          return 1
```

Comment:
@noahmmcgivern could you toggle those return values to match the semantic of the function name "isUserAdmin"? Given the phrasing I would expect this function to return 1 if the user is an admin.